File: theseus_gui/src/helpers/metadata.js
import { invoke } from '@tauri-apps/api/tauri'

/// Gets the game versions from daedalus
// Returns a VersionManifest
export async function get_game_versions() {
  return await invoke('metadata_get_game_versions')
}

// Gets the fabric versions from daedalus
// Returns Manifest
export async function get_fabric_versions() {
  return await invoke('metadata_get_fabric_versions')
}

// Gets the forge versions from daedalus
// Returns Manifest
export async function get_forge_versions() {
  return await invoke('metadata_get_forge_versions')
}

// Gets the quilt versions from daedalus
// Returns Manifest
export async function get_quilt_versions() {
  return await invoke('metadata_get_quilt_versions')
}

File: apps/theseus_gui_rs/src/api/metadata.rs
use crate::api::Result;
use daedalus::minecraft::VersionManifest;
use daedalus::modded::Manifest;

pub fn init<R: tauri::Runtime>() -> tauri::plugin::TauriPlugin<R> {
    tauri::plugin::Builder::new("metadata")
        .invoke_handler(tauri::generate_handler![
            metadata_get_game_versions,
            metadata_get_fabric_versions,
            metadata_get_forge_versions,
            metadata_get_quilt_versions,
            metadata_get_neoforge_versions,
        ])
        .build()
}

/// Gets the game versions from daedalus
#[tauri::command]
pub async fn metadata_get_game_versions() -> Result<VersionManifest> {
    Ok(theseus::metadata::get_minecraft_versions().await?)
}

/// Gets the fabric versions from daedalus
#[tauri::command]
pub async fn metadata_get_fabric_versions() -> Result<Manifest> {
    Ok(theseus::metadata::get_fabric_versions().await?)
}

/// Gets the forge versions from daedalus
#[tauri::command]
pub async fn metadata_get_forge_versions() -> Result<Manifest> {
    Ok(theseus::metadata::get_forge_versions().await?)
}

/// Gets the quilt versions from daedalus
#[tauri::command]
pub async fn metadata_get_quilt_versions() -> Result<Manifest> {
    Ok(theseus::metadata::get_quilt_versions().await?)
}

/// Gets the quilt versions from daedalus
#[tauri::command]
pub async fn metadata_get_neoforge_versions() -> Result<Manifest> {
    Ok(theseus::metadata::get_neoforge_versions().await?)
}
